### PR TITLE
MTDSA-13789: Endpoint code for Retrieve an Employment and its Financial Details TYS

### PR DESCRIPTION
**Pull request by SamuelJacobsHMRC**

### CODE REVIEW

File: test/v1/connectors/RetrieveEmploymentAndFinancialDetailsConnectorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        "the connector sends a request with a tax year parameter provided is supplied" in new TysIfsTest with Test {
+          override def taxYear: TaxYear = TaxYear.fromMtd("2023-24")
+          val expected                  = Right(ResponseWrapper(correlationId, response))
+
+          stubTysHttpResponse(expected)
+
+          await(connector.retrieve(request)) shouldBe expected
+        }
+
+        "a valid non-TYS request with all parameters provided is supplied" in new TysIfsTest with Test {
```
Comment (sean-flynn-HMRC) [inline test/v1/connectors/RetrieveEmploymentAndFinancialDetailsConnectorSpec.scala:112]: I think you mean to call this test "a valid TYS request..." The non-TYS scenarios are covered above

File: test/v1/connectors/RetrieveEmploymentAndFinancialDetailsConnectorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+        "a valid non-TYS request with a income source type parameter provided is supplied" in new Release6Test with Test {
+          val expected = Right(ResponseWrapper(correlationId, response))
+
+          stubHttpResponse(expected)
+
+          await(connector.retrieve(request)) shouldBe expected
+        }
+
+        "a valid non-TYS request with all parameters provided is supplied" in new Release6Test with Test {
```
Comment (sean-flynn-HMRC) [inline test/v1/connectors/RetrieveEmploymentAndFinancialDetailsConnectorSpec.scala:80]: Despite having different titles, the tests on line 56, 64, 72 and 80 in practice appear to be identical. 

Should these tests be different and if not, why are we executing the same test 4 times?

File: test/v1/requestParsers/validators/validations/EmploymentSourceValidationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  case class SetUp(employmentId: String)
+
+  "validate" should {
+    "return no errors" when {
+      "a valid employment source hmrcHeld is provided" in new SetUp("hmrcHeld") {
+
+        EmploymentSourceValidation.validate(employmentId).isEmpty shouldBe true
+      }
+      "a valid employment source user is provided" in new SetUp("user") {
```
Comment (sean-flynn-HMRC) [inline test/v1/requestParsers/validators/validations/EmploymentSourceValidationSpec.scala:33]: According to the old tech spec this is correct. But in the new tech spec (#1877) the only three possible values for source are 1) HMRC Held 2) CUSTOMER 3) Latest. 

Might be worth checking with a BA to confirm that "user" is being replaced by "customer" for the TYS update. If yes, then please add a test for this.

Comment (SamuelJacobsHMRC) [reply test/v1/requestParsers/validators/validations/EmploymentSourceValidationSpec.scala:33]: It should be user on the front-end and that is mapped to be customer for the back-end so no change between TYS and non-TYS

File: test/v1/connectors/RetrieveEmploymentAndFinancialDetailsConnectorSpec.scala (hunk 1 above)
Comment (SamuelJacobsHMRC) [reply test/v1/connectors/RetrieveEmploymentAndFinancialDetailsConnectorSpec.scala]: updated ✅

File: test/v1/connectors/RetrieveEmploymentAndFinancialDetailsConnectorSpec.scala (hunk 2 above)
Comment (SamuelJacobsHMRC) [reply test/v1/connectors/RetrieveEmploymentAndFinancialDetailsConnectorSpec.scala]: updated ✅

**Review by sean-flynn-HMRC — APPROVED** [review]
LGTM

**Pull request merged**